presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1424,6 +1436,39 @@ else if (materializedViewStatus.isPartiallyMaterialized()) {
             return SqlFormatterUtil.getFormattedSql(unionQuery, sqlParser, Optional.empty());
         }
 
+        /**
+         * Returns materialized view status, considering filter conditions from query
+         */
+        private MaterializedViewStatus getMaterializedViewStatus(QualifiedObjectName materializedViewName, Optional<Scope> scope)
+        {
+            TupleDomain<String> baseQueryDomain;
+            // We can use filter from base query WHERE clause to consider only relevant partitions when getting
+            // materialized view status
+            // TODO: Collect predicates in outer queries and apply the union to this criteria. i.e. in
+            //  SELECT x FROM (SELECT x FROM tbl WHERE y < 20) WHERE y > 5 where y is a partition key,
+            //  right now we would only remove partitions y >= 20. We should eventually consider y <= 5 as well.
+            if (currentSubquery.getWhere().isPresent() && isMaterializedViewPartitionFilteringEnabled(session)) {
+                Scope sourceScope = analyzeFrom(currentSubquery, scope);
+                Expression baseQueryWhereClause = currentSubquery.getWhere().get();
+
+                analyzeWhere(currentSubquery, sourceScope, baseQueryWhereClause);
+
+                DomainTranslator domainTranslator = new RowExpressionDomainTranslator(metadata);
+                RowExpression rowExpression = SqlToRowExpressionTranslator.translate(
+                        baseQueryWhereClause,
+                        analysis.getTypes(),
+                        ImmutableMap.of(),
+                        metadata.getFunctionAndTypeManager(),
+                        session);
+
+                baseQueryDomain = MaterializedViewUtils.getDomainFromFilter(session, domainTranslator, rowExpression);
+            }
+            else {
+                baseQueryDomain = TupleDomain.all();
+            }
+            return metadata.getMaterializedViewStatus(session, materializedViewName, baseQueryDomain);
```

Comment:
[Major] Once we're here, the query is from a materialized view. So `baseQueryDomain` variable contains predicates on materialized view, and not the base table. We need to do some work to convert these predicates to that of base table